build.gradle

```diff
@@ -255,6 +255,33 @@ model {
                 }
                 targetPlatform p.name
             }
+            binaries.all {
+                if (targetPlatform.operatingSystem.macOsX
+                    || targetPlatform.operatingSystem.linux) {
+                    cppCompiler.args "-g"                       // Produce debug output
+                    cppCompiler.args "-pthread"                 // Force nicer threading
+                    cppCompiler.args "-pedantic"                // Disable non-standard things
+                    cppCompiler.args "--std=c++11"              // Enable C++11
+                    cppCompiler.args "-Wall"                    // All warnings
+                    cppCompiler.args "-Wextra"                  // Plus extra
+                    cppCompiler.args "-Wformat=2"               // Check printf format strings
+                    cppCompiler.args "-Werror"                  // Warnings are errors
+                    cppCompiler.args "-Wno-format-nonliteral"   // Allow printf to have dynamic format string
+                    linker.args "-pthread"
+                } else if (targetPlatform.operatingSystem.windows) {
+                    cppCompiler.args "/DEBUG"                   // Produce debug output
+                    cppCompiler.args "/std:c++17"               // Won't hurt
```

Comment:
I think we should use the same `c++` version for Windows and Unix OSes.

Reply:
You can't, it's C++14 by default on Windows, and you can't do C++11. We don't want to go above C++11 with Linux or macOS, though.